listenstore/listenstore/listen.py

```diff
@@ -9,7 +9,7 @@ def __init__(self, user_id=None, timestamp=None, artist_msid=None, album_msid=No
                  recording_msid=None, data=None):
         self.user_id = user_id
         self.timestamp = timestamp
-        self.ts_since_epoch = calendar.timegm(self.timestamp.utctimetuple())
+        self.ts_since_epoch = calendar.timegm(self.timestamp.utctimetuple()) if self.timestamp else None
```

Comment:
There are 2 things here, let's address them one at a time: 

the `Listen` object has always had `from_json` as a static method (e.g., [1 year ago it was there](https://github.com/metabrainz/listenbrainz-server/blob/930b50f51c0623b030434485f675a58ed0d6e096/listenstore/listenstore/listen.py#L20)), so let's use that to fix this problem. 
Putting this change here doesn't sound like the easiest solution to me. This goes back to the question of talking about meaning. What does it mean to have a Listen with no timestamp? with no other data? It sounds like an invalid  state. we have a way to create an object from a json block without running into this bug, so let's do that.

Regarding playing now Listens, how about we have a separate object which doesn't require a timestamp? We can have a Listen and PlayingNow. Perhaps they can share a common base class for their shared fields, but then Listen has this extra field for the timestamp, and PlayingNow has (...? I don't know. Perhaps a date which it expires at?)